### PR TITLE
Add LTI env vars and django secure settings

### PR DESCRIPTION
**Pull request by joshuagetega**
This PR adds the LTI credentials needed to install hedera in Canvas.
These include the "consumer key" and "shared secret".
The credentials are read from the ECS environment in which the application is deployed and passed into gunicorn.
Also added are django secure settings.

@arthurian @dodget

### CODE REVIEW

**Review by dodget — COMMENTED** [review]
@joshuagetega - There is also the `IS_LTI` env var to be aware of in the settings file:
`IS_LTI = (os.environ.get("IS_LTI") == "1")`
I don't think the lti will work correctly unless that value is set in the environment

**Comment by joshuagetega** [comment]
> @joshuagetega - There is also the `IS_LTI` env var to be aware of in the settings file:
> `IS_LTI = (os.environ.get("IS_LTI") == "1")`
> I don't think the lti will work correctly unless that value is set in the environment

Good catch, @dodget. I have added it and force-pushed.

**Review by dodget — APPROVED** [review]
Looks good @joshuagetega

File: hedera/settings.py [hunk 1]
```diff
@@ -8,7 +8,10 @@
 from .aws import get_ecs_task_ips
 
 
-IS_LTI = (os.environ.get("IS_LTI") == "1")
+try:
+    IS_LTI = bool(int(os.environ.get("IS_LTI", "0")))
```
Comment (arthurian) [inline hedera/settings.py:12]: Good catch on the cast to `int` and corresponding try/catch. Otherwise you'd have to use an empty string to get `bool` to evaluate to false.

File: hedera/settings.py [hunk 2]
```diff
@@ -406,3 +409,8 @@ def user_display(user):
 SESSION_COOKIE_SAMESITE = None
 
 WAGTAIL_SITE_NAME = "Hedera"
+
+
+# SSL is terminated at the ELB so look for this header to know that we should be in ssl mode
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+SESSION_COOKIE_SECURE = True
```
Comment (arthurian) [inline hedera/settings.py:416]: Good catch on the SSL related settings.

File: lti/views.py [hunk 3]
```diff
@@ -38,25 +40,30 @@ def dispatch(self, request, *args, **kwargs):
...
+        #     return render(request, "lti_failure-verify.html")
 
         if not request.user.is_authenticated:
             try:
                 lti_user = login_existing_user(request)
             except EmailAddress.DoesNotExist:
                 lti_email = request.POST.get(LTI_PROPERTY_USER_EMAIL, None)
                 if not lti_email:
-                    return render(request, "lti_failure.html")
+                    logger.debug("The error is there is no lti_email")
```
Comment (arthurian) [inline lti/views.py:56]: I would suggest using `logger.error()` here since this might indicate a problem with the tool configuration if the email isn't coming through in the LTI launch and it's unlikely the user would be able to do anything to resolve it themselves. We'd want to see this in the logs. I think the subsequent errors would also benefit from being categorized similarly.

**Review by arthurian — APPROVED** [review]
@joshuagetega The PR looks good overall. I do have a few questions about some of the changes that are separate from setting the consumer key/secret.

File: lti/views.py [hunk 4]
```diff
@@ -38,25 +40,30 @@ def dispatch(self, request, *args, **kwargs):
 
         # validate the request
         lti = LTI(request_type="any", role_type="any")
-        try:
-            lti.verify(request)
-        except LTIException:
-            return render(request, "lti_failure.html")
+        lti.verify(request)
```
[the remark this answers is not in the document]
Comment (dodget) [reply lti/views.py:43]: That was done to let the exception raise so we could do some debugging. I think Josh's plan is to revert this and issue a fresh PR

[the remark this answers is not in the document]
Comment (arthurian) [reply lti/views.py:43]: Makes sense, thanks!

File: lti/views.py [hunk 5]
```diff
@@ -38,25 +40,30 @@ def dispatch(self, request, *args, **kwargs):
...
 
         if not request.user.is_authenticated:
             try:
                 lti_user = login_existing_user(request)
             except EmailAddress.DoesNotExist:
                 lti_email = request.POST.get(LTI_PROPERTY_USER_EMAIL, None)
                 if not lti_email:
-                    return render(request, "lti_failure.html")
+                    logger.debug("The error is there is no lti_email")
+                    return render(request, "lti_failure-email.html")
```
[the remark this answers is not in the document]
Comment (dodget) [reply lti/views.py:57]: Yeah, totally. This was just something I added in quickly to debug the lti config issue we were seeing. This will also be reverted to the way it was, I believe, before the PR is supposed to be ready for review

**Comment by joshuagetega** [comment]
@arthurian, our bad we should have looped you in on this. The LTI wasn't working in dev at first, and @dodget and I couldn't crack it right away. To figure it out, we had to do temporary things like [remove the handling of an exception](https://github.com/Hedera-Lang-Learn/hedera/pull/272/commits/c5cafbcc7110d6e1fcc902eb63d672e1454c850b) so we could see the error message being thrown, add some logging etc... Eventually we realized the problem was [ssl settings](https://github.com/Hedera-Lang-Learn/hedera/pull/272/commits/65397896e87724e194aa40c2b379b0b4d273ab16) were missing in the settings file.

In a nutshell, the PR isn't ready for review. I'm gonna clean it up and undo some of the temporary changes we added. But the comments are helpful already

**Comment by arthurian** [comment]
@joshuagetega My bad! I jumped the gun :) Feel free to ignore my comments until it's finalized.

**Comment by joshuagetega** [comment]
Cool cool, @arthurian . No worries :)

**Comment by joshuagetega** [comment]
@arthurian @dodget, I have reverted the temporary changes. This is ready for review now. I have tested it in dev, i.e. the harvard-hedera-dev (lti) ECS service, and successfully done an LTI launch from Canvas.

**Comment by paltman** [comment]
@joshuagetega I added the things we talked about on the call today here

**Comment by paltman** [comment]
is this ready?

**Comment by joshuagetega** [comment]
Not yet, @paltman. I am going to split the `settings.py` module into 3. Namely, a `base.py` that contains settings common to all local/dev/prod environments, a `local.py` that extends `base.py` and adds local settings, and an `aws.py` that extends `base.py` and adds aws-specific settings.

**Comment by joshuagetega** [comment]
> Not yet, @paltman. I am going to split the `settings.py` module into 3. Namely, a `base.py` that contains settings common to all local/dev/prod environments, a `local.py` that extends `base.py` and adds local settings, and an `aws.py` that extends `base.py` and adds aws-specific settings.

Changed my mind. I thought adding `SECURE_PROXY_SSL_HEADER` and `SESSION_COOKIE_SECURE` would break the local development environment. But upon testing, the local dev environment works fine even with the those additional settings. So I am going to leave the `settings.py` module as is. Changing it unnecessarily will make the project more complex than it needs to be. So, I am going ahead with the merge of this PR.
